webpack/worker/worker_adapter.js

```diff
@@ -28,118 +81,103 @@ function main() {
   }
 }
 
-/**
- * Returns arguments that is preceded by -c
- * @param {string[]} args
- * @returns {string[]}
- */
-function findWebpackConfigs(args) {
-  const configs = []
-  for (let i = 0; i < args.length; i++) {
-    if (i > 0 && args[i - 1] == '-c') {
-      configs.push(args[i])
-    }
-  }
-  return configs
-}
-
-function runAsPersistentWorker() {
-  const webpackCliPath = require.resolve('webpack-cli/bin/cli.js')
-  /** @type {string} */
-  let key
-  /** @type {cp.ChildProcess} */
-  let proc
-
-  /** @type {Map<string, string>} */
-  let configDigestMap = new Map()
+async function runAsPersistentWorker() {
+  const cli = new WorkerAwareCLI()
+  /** @type {import("@bazel/worker").Inputs} */
+  let inputMap
+  const rootPath = process.cwd()
+  const execrootNodeModules = path.join(rootPath, 'node_modules')
 
   /**
    * @param args {string[]}
    * @param inputs { [path: string]: string }
    */
   const build = async (args, inputs) => {
-    return new Promise((resolve, reject) => {
-      const configs = findWebpackConfigs(args)
-
-      for (const config of configs) {
-        if (configDigestMap.get(config) != inputs[config]) {
-          configDigestMap.set(config, inputs[config])
-          if (proc && !proc.killed) {
-            console.error(
-              `a config file change ${config} has been detected. Restarting webpack..`
-            )
-            proc.kill()
-          }
-        }
-      }
-
-      // We can not add --watch argument earlier in the starlark side
-      // until we know for sure which mode we are working on. in RBE
-      // local execution strategy will be the default and we have
-      // no choice but to add it dynamically on runtime.
-      args = [...args, '--watch']
-
-      const argumentKey = args.join('#')
-
-      if (key != argumentKey) {
-        worker.log(`Arguments have changed. Killing the process.`)
-        proc?.kill()
-      }
+    worker.log(`[${MNEMONIC}] Building`)
+    if (!fs.existsSync(execrootNodeModules)) {
+      worker.log(`[${MNEMONIC}] execroot/node_modules is missing relinking.`)
```

Comment:
What does `relinking` mean (if I was a user would I know that this meant)? Could you rework this log? Here and below

Reply:
Hmm, you are right. users already have no idea what the linker does. Do you have a suggestion about how I should form this sentence?

Reply:
relinking in this context means; we are running the linker (which is part of the original launcher) again to mutate execroot to put node_modules for module resolution because bazel removed it.